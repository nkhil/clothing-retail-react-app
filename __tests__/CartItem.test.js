import "@babel/polyfill"
import React from "react"
import ShoppingCartItem from "../src/App/components/CartItem"
import renderer from "react-test-renderer"
import ReactDOM from "react-dom"

require("react-router-dom")
jest.mock("react-router-dom")

const mockProduct = {
  productName: "Almond Toe Court Shoes, Patent Black",
  productCategory: "Women's Footwear",
  productPrice: 9900,
  productQuantity: 5,
}

const mockRemoveFromCart = () => {}

test("ShoppingCartItem renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(
    <ShoppingCartItem
      key={"product1"}
      index={"product1"}
      numberOfItems={1}
      product={mockProduct}
      removeFromCart={mockRemoveFromCart}
    />,
    div
  )
})

test("ShoppingCart renders correctly", () => {
  const shoppingCartItem = renderer
    .create(
      <ShoppingCartItem
        key={"product1"}
        index={"product1"}
        numberOfItems={1}
        product={mockProduct}
        removeFromCart={mockRemoveFromCart}
      />
    )
    .toJSON()
  expect(shoppingCartItem).toMatchSnapshot()
})
