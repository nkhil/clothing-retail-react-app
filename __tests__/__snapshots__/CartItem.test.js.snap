// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart renders correctly 1`] = `
<li
  className="product"
  id="product1"
>
  <h3>
    Almond Toe Court Shoes, Patent Black
     x 
    1
  </h3>
  <button
    className="remove-item"
    onClick={[Function]}
  >
    Remove
  </button>
</li>
`;
